www-misc/buku/buku-4.3.ebuild

```diff
@@ -0,0 +1,66 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+DISTUTILS_USE_SETUPTOOLS=rdepend
+
+inherit bash-completion-r1 distutils-r1
+
+DESCRIPTION="Powerful command-line bookmark manager"
+HOMEPAGE="https://github.com/jarun/buku"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-python/html5lib-1.0.1[${PYTHON_USEDEP}]
+	>=dev-python/beautifulsoup-4.6.0[${PYTHON_USEDEP}]
+	dev-python/cryptography[${PYTHON_USEDEP}]
+	dev-python/urllib3[${PYTHON_USEDEP}]
+"
+
+DEPEND="${RDEPEND}
+	test? (
+		>=dev-python/click-7.0[${PYTHON_USEDEP}]
+		>=dev-python/py-1.5.0[${PYTHON_USEDEP}]
+		>=dev-python/pylint-1.7.2[${PYTHON_USEDEP}]
+		>=dev-python/pyyaml-4.2[${PYTHON_USEDEP}]
+		>=dev-python/vcrpy-4.0.2[${PYTHON_USEDEP}]
+		dev-python/attrs[${PYTHON_USEDEP}]
+		dev-python/flake8[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+	)
+"
+
+python_prepare() {
+	distutils-r1_python_prepare_all
+
+	# Remove support for bukuserver - complex depgraph which isn't all
+	# sufficiently packaged in Gentoo
+	sed -ie '/console_scripts/s/,.*/]/' setup.py || die
+}
+
+python_install_all() {
+	distutils-r1_python_install_all
+
+	insinto /usr/share/zsh/site-functions
+	doins auto-completion/zsh/_*
+
+	newbashcomp auto-completion/bash/buku-completion.bash "${PN}"
+
+	doman buku.1
+}
+
+python_test() {
+	# Explicitly enumerate tests to avoid tests that deal with bukuserver
+	tests=$(echo "tests/test_"{BukuCrypt,bukuDb,buku,ExtendedArgumentParser,import_firefox_json}".py")
+
+	py.test -v ${tests} || die
```

Comment:
Is the variable needed here? I think you can just specify these files after `py.test`.

```
========================================== ERRORS ==========================================
____________________ ERROR at setup of test_load_firefox_database[True] ____________________

self = <urllib.request.HTTPSHandler object at 0x7f8442cde2e8>
http_class = <class 'http.client.HTTPSConnection'>
req = <urllib.request.Request object at 0x7f84422f2898>
http_conn_args = {'check_hostname': None, 'context': None}, host = 'github.com'
h = <http.client.HTTPSConnection object at 0x7f8442cde358>
...
...
tests/test_bukuDb.py:1190: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python3.6/urllib/request.py:223: in urlopen
    return opener.open(url, data, timeout)
/usr/lib64/python3.6/urllib/request.py:526: in open
    response = self._open(req, data)
/usr/lib64/python3.6/urllib/request.py:544: in _open
    '_open', req)
/usr/lib64/python3.6/urllib/request.py:504: in _call_chain
    result = func(*args)
/usr/lib64/python3.6/urllib/request.py:1361: in https_open
    context=self._context, check_hostname=self._check_hostname)
...
...
tests/test_bukuDb.py:627: AssertionError
----------------------------------- Captured stderr call -----------------------------------
WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8442c930f0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com
WARNING:urllib3.connectionpool:Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8442c931d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com
WARNING:urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8442ec3128>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com
ERROR:root:HTTPSConnectionPool(host='tny.im', port=443): Max retries exceeded with url: /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f84431e6160>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',))
```
etc, could you get tests to run? `========== 17 failed, 344 passed, 2 skipped, 4 warnings, 2 error in 2.78 seconds ===========`

Reply:
The tests work with FEATURES=-network-sandbox. It uses vcrpy but seemingly not for everything. I can ask upstream to fix it.

Reply:
I guess you'll need to restrict tests then for this release.

Reply:
I have an [open PR](https://github.com/jarun/buku/pull/443) and [open issue](https://github.com/jarun/buku/issues/442) upstream to fix it, but they've proven pretty hostile to network-sandbox.